### PR TITLE
Simplification and type improvement in button component

### PR DESCRIPTION
**Pull request by esanzgar**
IconButton should no have children props.

Depends on #22

### CODE REVIEW

**Comment by lyzadanger** [comment]
@esanzgar Do you want to rebase to fix conflicts and then I'll review?

**Comment by esanzgar** [comment]
Fixed the conflict.

File: src/components/buttons.js [hunk 1]
```diff
@@ -36,34 +37,18 @@ import { SvgIcon } from './SvgIcon';
...
-    otherAttributes['aria-label'] = title;
-  }
-
-  if (typeof expanded === 'boolean') {
-    otherAttributes['aria-expanded'] = expanded;
-  }
-  if (typeof pressed === 'boolean') {
-    otherAttributes['aria-pressed'] = pressed;
-  }
+  restProps['aria-expanded'] = expanded;
```
Comment (esanzgar) [inline src/components/buttons.js:49]: If `expanded` is undefined the prop is ignored

File: src/components/test/buttons-test.js [hunk 2]
```diff
@@ -124,6 +124,24 @@ function addCommonTests({ componentName, createComponentFn, withIcon = true }) {
         attributeName: 'disabled',
         attributeValue: '',
       },
+      {
+        propName: 'expanded',
```
Comment (esanzgar) [inline src/components/test/buttons-test.js:128]: This test shows that for undefined prop the attribute is not set.

File: src/components/buttons.js [hunk 3]
```diff
@@ -26,6 +26,7 @@ import { SvgIcon } from './SvgIcon';
  * @typedef IconButtonBaseProps
  * @prop {string} icon - Icon is required for icon buttons
  * @prop {string} title - Title is required for icon buttons
+ * @prop {never} [children] - children are not allowed (use LabeledButton instead)
```
Comment (lyzadanger) [inline src/components/buttons.js:29]: This is so handy! I had been wondering how to "remove" a property from a typedef and hadn't been able to google this up.

File: src/components/buttons.js (hunk 1 above)
Comment (lyzadanger) [reply src/components/buttons.js:49]: Thank you; much simpler.

File: src/components/buttons.js [hunk 4]
```diff
@@ -36,34 +37,18 @@ import { SvgIcon } from './SvgIcon';
...
-  expanded,
-  pressed,
-  onClick = () => {},
   size = 'medium',
-  style = {},
-  title,
   variant = 'normal',
+  expanded,
+  pressed,
+  ...restProps
```
Comment (lyzadanger) [inline src/components/buttons.js:47]: I wonder if hiding, e.g., `onClick` in these `restProps` makes it a little bit opaque...

File: src/components/buttons.js [hunk 5]
```diff
@@ -89,9 +70,8 @@ function ButtonBase({
  *
  * @param {IconButtonProps} props
  */
-export function IconButton(props) {
-  const { className = 'IconButton', ...restProps } = props;
-  const { icon } = props;
+export function IconButton({ className = 'IconButton', ...restProps }) {
```
Comment (lyzadanger) [inline src/components/buttons.js:73]: I definitely like the `restProps` usage here 👍🏻

**Review by lyzadanger — APPROVED** [review]
With the caveat that I cannot preview these buttons visually in this package until we've got the pattern library in place (it's on the list!), these changes look good.

The only thing I wasn't 100% sure about was hiding meaningful props in `ButtonBase` in `...restProps`, e.g. the `onClick` property, which is rather fundamental. Using props spread for properties that are just "passing through" components seems like a nice pattern, I'm just wondering if we should stay more explicit in the components where they're actually _used_. Not a strong opinion.

File: src/components/buttons.js (hunk 3 above)
Comment (esanzgar) [reply src/components/buttons.js:29]: I was recently reminded about `never` watching this video: https://www.youtube.com/watch?v=vXh4PFwZFGI

File: src/components/buttons.js (hunk 4 above)
Comment (esanzgar) [reply src/components/buttons.js:47]: Personally, I think it makes a better distinction between the props that are allowed but not directly used or manipulated by `ButtonBase` (`children`, `disabled`, `onClick`, `style`, `title`), which go to `restProps`, from the props that are exclusive of `ButtonBase` (`className`, `icon`, `iconPosition`, `expanded`, `pressed`, `size`)

**Pull request merged**